builtin/common/vector.lua

```diff
@@ -141,3 +141,95 @@ function vector.sort(a, b)
 	return {x = math.min(a.x, b.x), y = math.min(a.y, b.y), z = math.min(a.z, b.z)},
 		{x = math.max(a.x, b.x), y = math.max(a.y, b.y), z = math.max(a.z, b.z)}
 end
+
+local function sin(x)
+	if x % math.pi == 0 then
+		return 0
+	else
+		return math.sin(x)
+	end
+end
```

Comment:
This feels like the wrong approach to ensure that results are exact

Reply:
p_gimeno once said that special casing is better than rounding: http://irc.minetest.net/minetest-dev/2019-05-11#i_5542561

Reply:
A comment should be added here to explain the situation around rounding/exactness then.